src/url-state-machine.js

```diff
@@ -106,19 +107,19 @@ function utf8PercentEncode(c) {
 }
 
 function utf8PercentDecode(str) {
-  const input = new Buffer(str);
+  const input = utf8.encode(str);
   const output = [];
   for (let i = 0; i < input.length; ++i) {
     if (input[i] !== p("%")) {
       output.push(input[i]);
     } else if (input[i] === p("%") && isASCIIHex(input[i + 1]) && isASCIIHex(input[i + 2])) {
-      output.push(parseInt(input.slice(i + 1, i + 3).toString(), 16));
+      output.push(parseInt(utf8.decode(input.slice(i + 1, i + 3)), 16));
```

Comment:
Yes, and I already gave you a solution, which you didn't implement for unknown reasons.